model/perf.go

```diff
@@ -436,3 +436,42 @@ func (r *PerformanceResults) findAllChildrenGraphLookup(parent string, maxDepth
 	}
 	return nil
 }
+
+// Returns performance results with missing or outdated rollup information for
+// the given `name` and `version`.
+func (r *PerformanceResults) FindOutdatedRollups(name string, version int, after time.Time) error {
+	conf, session, err := cedar.GetSessionWithConfig(r.env)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer session.Close()
+
+	search := bson.M{
+		perfCreatedAtKey: bson.M{"$gt": after},
+		bsonutil.GetDottedKeyName(perfArtifactsKey, artifactInfoFormatKey): FileFTDC,
+		"$or": []bson.M{
+			{
+				bsonutil.GetDottedKeyName(perfRollupsKey, perfRollupsStatsKey): bson.M{
+					"$elemMatch": bson.M{
+						perfRollupValueNameKey:    name,
+						perfRollupValueVersionKey: bson.M{"$lt": version},
+					},
+				},
+			},
+			{
+				bsonutil.GetDottedKeyName(perfRollupsKey, perfRollupsStatsKey, perfRollupValueNameKey): bson.M{"$ne": name},
+			},
+		},
+	}
+	err = session.DB(conf.DatabaseName).C(perfResultCollection).Find(search).All(&r.Results)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"problem finding perf results with outdated rollup %s (version %d)",
+			name,
+			version,
+		)
```

Comment:
I'd combine this on one line. To be honest you can really combine 466-476 on one or two lines, (but one statement)